src/libraries/Native/Unix/System.Native/pal_networking.c

```diff
@@ -2297,6 +2418,48 @@ int32_t SystemNative_Socket(int32_t addressFamily, int32_t socketType, int32_t p
     return Error_SUCCESS;
 }
 
+int32_t SystemNative_GetSocketType(intptr_t socket, int32_t* addressFamily, int32_t* socketType, int32_t* protocolType)
+{
+    if (addressFamily == NULL || socketType == NULL || protocolType == NULL)
+    {
+        return Error_EFAULT;
+    }
+
+    int fd = ToFileDescriptor(socket);
+
+#ifdef SO_DOMAIN
+    sa_family_t domainValue;
```

Comment:
Linux considers this an `int`. I think this won't work on a big-endian system because it will copy the two (`sizeof(sa_family_t`)  most significant bytes of the `int`, which will be zero.

Reply:
I'm not following.  Are you saying Linux considers sa_family_t to be four-bytes, but getsockopt on Linux doesn't consider SO_DOMAIN to be four-bytes and will only populate two?

Reply:
Linux `sizeof(sa_family_t)` is `2`.
Value returned for `SO_DOMAIN` is `sizeof(int)`.
Because the provided buffer is too small, it will crop the returned value to the first two bytes.
That will work on little-endian, but not on big-endian.

Reply:
> Linux sizeof(sa_family_t) is 2.
> Value returned for SO_DOMAIN is sizeof(int).

That's messed up :)  But ok, will fix.